spec/models/guest_room_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe GuestRoom, type: :model do
+  it {should belong_to :guest}
+  it {should belong_to :room }
+
+  describe 'instance methods' do
+    describe '#reservation_count' do
+      let!(:holliday) { Hotel.create!(name: 'Holliday Inn', location: 'Buxton') }
+      let!(:harbor) { Hotel.create!(name: 'Portland Harbor', location: 'Portland') }
+      let!(:room_1) { harbor.rooms.create!(rate: 180, suite: "Presidential")}
+      let!(:room_2) { harbor.rooms.create!(rate: 185, suite: "Vice-Presidential")}
+      let!(:room_3) { holliday.rooms.create!(rate: 160, suite: "Honeymoon")}
+      let!(:room_4) { holliday.rooms.create!(rate: 10, suite: "Broom Closet")}
+      let!(:steve) { Guest.create!(name: 'Steve', nights: 3) }
+      let!(:dave) { Guest.create!(name: 'Dave', nights: 5) }
+      let!(:jim) { Guest.create!(name: 'Jim', nights: 1)}
+
+  before do
+    GuestRoom.create!(guest: steve, room: room_1)
+    GuestRoom.create!(guest: steve, room: room_2)
+    GuestRoom.create!(guest: dave, room: room_1)
+    GuestRoom.create!(guest: dave, room: room_3)
+    GuestRoom.create!(guest: dave, room: room_4)
+  end
+
+      it 'counts the number of guests who have rented the room' do
+        expect(room_1.reservations).to eq(2)
+      end
+    end
+  end
+end
```

Comment:
I think we are overcomplicating things here a bit. I think what we want is an instance method on the `Room` model that counts all the guests through the association it has with the GuestRooms table.

On your `Room` model:

```ruby
def guest_count
  guests.count
end
```

And that would have been all you needed. There may be good reasoning behind adding a method to the join model, but I don't think this was one of them.